### PR TITLE
Update to MatchEvaluator NRE change

### PR DESCRIPTION
**Pull request by danmosemsft**
Avoid calling ValueStringBuilder at all. Slightly shorter code path.

### CODE REVIEW

**Comment by danmosemsft** [comment]
https://github.com/dotnet/corefx/pull/39779

File: src/System.Text.RegularExpressions/src/System/Text/RegularExpressions/Regex.Replace.cs [hunk 1]
```diff
@@ -186,7 +186,9 @@ private static string Replace(MatchEvaluator evaluator, Regex regex, string inpu
                             vsb.Append(input.AsSpan(prevat, match.Index - prevat));
 
                         prevat = match.Index + match.Length;
-                        vsb.Append(evaluator(match) ?? "");
+                        string result = evaluator(match);
+                        if (!String.IsNullOrEmpty(result))
```
Comment (stephentoub) [inline src/System.Text.RegularExpressions/src/System/Text/RegularExpressions/Regex.Replace.cs]: Nit: String => string

Comment (danmosemsft) [reply src/System.Text.RegularExpressions/src/System/Text/RegularExpressions/Regex.Replace.cs]: oops. My last team had the opposite coding style..

**Comment by danmosemsft** [comment]
macOS queue is flaky. Everything else passed and it previously passed on macOS before I changed "S" to "s". merging.

**Comment by MattGal** [comment]
@danmosemsft we've had some issues with the core dump support filling up disks and I'm monitoring this.  The problem is that whichever mac is causing the problem is not logging its machine name in the messages it sends.

**Comment by danmosemsft** [comment]
thanks @mattgal. I didn't even summon you!

**Comment by MattGal** [comment]
Toub pointed me at the issues this morning the moment I unlocked my workstation, as per usual.

I think the 10.13 stuff's fixed now, at least I can't find actively broken machines and PRs like this have green runs: https://dev.azure.com/dnceng/public/_build/results?buildId=281494

**Pull request merged**